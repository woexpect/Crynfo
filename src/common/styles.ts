import {Dimensions, Platform} from 'react-native';
import {ScaledSheet, verticalScale} from 'react-native-size-matters';

const {width, height} = Dimensions.get('window');

const isIosDevice = Platform.OS === 'ios';

export default ScaledSheet.create({
  // Layout Styles
  completeScreen: {
    flex: 1,
  },
  centerAll: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  completeWidth: {
    width: '100%',
  },
  cryptoListContainer: {
    width: '100%',
    height: verticalScale(64),
  },
  hideOverflow: {
    overflow: 'hidden',
  },
  centerVertically: {
    justifyContent: 'center',
  },
  spaceEvenly: {
    justifyContent: 'space-evenly',
  },
  spaceBetween: {
    justifyContent: 'space-between',
  },
  toEnd: {
    alignItems: 'flex-end',
  },
  roundedTopCorners: {
    borderTopRightRadius: verticalScale(16),
    borderTopLeftRadius: verticalScale(16),
  },
  onTop: {
    position: 'absolute',
  },
  cover: {
    width: '100%',
    height: '100%',
  },
  row: {
    flexDirection: 'row',
  },
  // Paddings
  paddingRight16: {
    paddingRight: verticalScale(16),
  },
  paddingHorizontal: {
    paddingHorizontal: verticalScale(16),
  },
  // Margins
  marginLeft16: {
    marginLeft: verticalScale(16),
  },
  marginRight16: {
    marginRight: verticalScale(16),
  },
  marginTop8: {
    marginTop: verticalScale(8),
  },
  marginTop1: {
    marginTop: 1,
  },
  marginTop16: {
    marginTop: verticalScale(16),
  },
  marginTop24: {
    marginTop: verticalScale(24),
  },
  // Background Colors
  bgBlack: {
    backgroundColor: '#141414',
  },
  bgMain: {
    backgroundColor: '#DEF263',
  },
  // Tint Colors
  tintBlack: {
    tintColor: '#141414',
  },
  tintMain: {
    tintColor: '#DEF263',
  },
  // Text Colors
  tcRed: {
    color: '#F26363',
  },
  tcBlack: {
    color: '#141414',
  },
  tcMain: {
    color: '#DEF263',
  },
  tcBlack8: {
    color: '#1414141F',
  },
  tcBlack64: {
    color: '#141414A3',
  },
  tcMain16: {
    color: '#DEF26329',
  },
  // Border Colors
  bcBlack: {
    borderColor: '#141414',
  },
  bcMain: {
    borderColor: '#DEF263',
  },
  // Font Families
  black: {
    fontFamily: 'Inter-Black',
  },
  bold: {
    fontFamily: 'Inter-Bold',
  },
  semiBold: {
    fontFamily: 'Inter-SemiBold',
  },
  medium: {
    fontFamily: 'Inter-Medium',
  },
  regular: {
    fontFamily: 'Inter-Regular',
  },
  // Font sizes
  cryptoPlaceholder: {
    fontSize: '112@ms0.3',
    includeFontPadding: false,
    lineHeight: '112@ms0.3',
    marginTop: -8,
    marginLeft: -4,
    letterSpacing: -2,
  },
  title: {
    fontSize: '26@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h1: {
    fontSize: '24@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h15: {
    fontSize: '22@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h2: {
    fontSize: '20@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  subtitle: {
    fontSize: '18@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largerText: {
    fontSize: '16@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largeText: {
    fontSize: '14@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  mediumText: {
    fontSize: '12@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  normalText: {
    fontSize: '10@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  smallText: {
    fontSize: '6@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
});
