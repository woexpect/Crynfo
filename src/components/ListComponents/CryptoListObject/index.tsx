import React from 'react';
import {Pressable, Text, View} from 'react-native';
// Custom components
// User tools
import styles from '../../../common/styles';

const CryptoListObject = ({
  addStyles,
  name = '',
  symbol = '',
  USDPrice = '$0',
  BTCRepresentation = '0.00BTC',
  isFirstItem = false,
}: {
  addStyles?: any;
  name: string;
  symbol: string;
  USDPrice: string;
  BTCRepresentation: string;
  isFirstItem?: boolean;
}) => {
  return (
    <Pressable
      style={[
        styles.cryptoListContainer,
        styles.hideOverflow,
        styles.bgMain,
        styles.centerVertically,
        isFirstItem && styles.roundedTopCorners,
        addStyles,
      ]}>
      <Text style={[styles.tcBlack8, styles.black, styles.cryptoPlaceholder]}>
        {symbol}
      </Text>
      <View
        style={[
          styles.onTop,
          styles.cover,
          styles.centerVertically,
          styles.row,
        ]}>
        <View style={[styles.completeScreen, styles.centerVertically]}>
          <Text
            style={[
              styles.tcBlack,
              styles.semiBold,
              styles.largerText,
              styles.marginLeft16,
            ]}>
            {name}
          </Text>
        </View>
        <View
          style={[
            styles.completeScreen,
            styles.spaceEvenly,
            styles.toEnd,
            styles.paddingRight16,
          ]}>
          <Text style={[styles.tcBlack, styles.semiBold, styles.largeText]}>
            {USDPrice}
          </Text>
          <Text style={[styles.tcBlack64, styles.semiBold, styles.largeText]}>
            {BTCRepresentation}
          </Text>
        </View>
      </View>
    </Pressable>
  );
};

export default React.memo(CryptoListObject);
