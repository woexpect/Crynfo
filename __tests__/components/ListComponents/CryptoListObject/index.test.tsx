import 'react-native';
import React from 'react';
import CryptoListObject from '../../../../src/components/ListComponents/CryptoListObject';

// Note: import explicitly to use the types shiped with jest.
import {expect, it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';

import styles from '../../../../src/common/styles';

it('renders correctly with no values', () => {
  const crypto = {
    id: 90,
    name: 'test crypto',
    symbol: 'BTC',
    price_usd: '1232134.33',
    price_btc: '12.123',
  };

  const action = () => {
    console.log('action called');
  };
  const tree = renderer.create(
    <CryptoListObject onPress={action} crypto={crypto} />,
  );

  expect(tree).toMatchSnapshot();
});

it('renders correctly with required values only', () => {
  const crypto = {
    id: 90,
    name: 'test crypto',
    symbol: 'BTC',
    price_usd: '1232134.33',
    price_btc: '12.123',
  };

  const action = () => {
    console.log('action called');
  };
  const tree = renderer.create(
    <CryptoListObject
      name={crypto.name}
      symbol={crypto.symbol}
      USDPrice={crypto.price_usd}
      BTCRepresentation={crypto.price_btc}
      onPress={action}
      crypto={crypto}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it('renders correctly with all possible values', () => {
  const crypto = {
    id: 90,
    name: 'test crypto',
    symbol: 'BTC',
    price_usd: '1232134.33',
    price_btc: '12.123',
  };

  const action = () => {
    console.log('action called');
  };
  const tree = renderer.create(
    <CryptoListObject
      addStyles={styles.marginTop24}
      name={crypto.name}
      symbol={crypto.symbol}
      USDPrice={crypto.price_usd}
      BTCRepresentation={crypto.price_btc}
      isFirstItem
      onPress={action}
      crypto={crypto}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it('Should call action with the required crypto object', () => {
  const crypto = {
    id: 90,
    name: 'test crypto',
    symbol: 'BTC',
    price_usd: '1232134.33',
    price_btc: '12.123',
  };

  const action = jest.fn((item: any) => {
    return item;
  });
  const tree = renderer.create(
    <CryptoListObject
      addStyles={styles.marginTop24}
      name={crypto.name}
      symbol={crypto.symbol}
      USDPrice={crypto.price_usd}
      BTCRepresentation={crypto.price_btc}
      isFirstItem
      onPress={action}
      crypto={crypto}
    />,
  );

  const listItem = tree.root.findByProps({
    testID: 'cryptoListObject',
  }).props;

  act(() => listItem.onPress());
  expect(action).toHaveBeenCalledWith(crypto);
});
