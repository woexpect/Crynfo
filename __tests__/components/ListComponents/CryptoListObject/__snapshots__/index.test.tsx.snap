// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all possible values 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "height": 125.55294117647058,
        "width": "100%",
      },
      {
        "overflow": "hidden",
      },
      {
        "backgroundColor": "#DEF263",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderTopLeftRadius": 31.388235294117646,
        "borderTopRightRadius": 31.388235294117646,
      },
      {
        "marginTop": 47.08235294117647,
      },
    ]
  }
  testID="cryptoListObject"
>
  <Text
    style={
      [
        {
          "color": "#1414141F",
        },
        {
          "fontFamily": "Inter-Black",
        },
        {
          "fontSize": 150.4,
          "includeFontPadding": false,
          "letterSpacing": -2,
          "lineHeight": 150.4,
          "marginLeft": -4,
          "marginTop": -8,
        },
      ]
    }
  >
    BTC
  </Text>
  <View
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": "100%",
          "width": "100%",
        },
        {
          "justifyContent": "center",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 21.485714285714284,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
            {
              "marginLeft": 31.388235294117646,
            },
          ]
        }
      >
        test crypto
      </Text>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "space-evenly",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "paddingRight": 31.388235294117646,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        $1232134.33
      </Text>
      <Text
        style={
          [
            {
              "color": "#141414A3",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        12.123 BTC
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with no values 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "height": 125.55294117647058,
        "width": "100%",
      },
      {
        "overflow": "hidden",
      },
      {
        "backgroundColor": "#DEF263",
      },
      {
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  testID="cryptoListObject"
>
  <Text
    style={
      [
        {
          "color": "#1414141F",
        },
        {
          "fontFamily": "Inter-Black",
        },
        {
          "fontSize": 150.4,
          "includeFontPadding": false,
          "letterSpacing": -2,
          "lineHeight": 150.4,
          "marginLeft": -4,
          "marginTop": -8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": "100%",
          "width": "100%",
        },
        {
          "justifyContent": "center",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 21.485714285714284,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
            {
              "marginLeft": 31.388235294117646,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "space-evenly",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "paddingRight": 31.388235294117646,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        $$0
      </Text>
      <Text
        style={
          [
            {
              "color": "#141414A3",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        0.00BTC BTC
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with required values only 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "height": 125.55294117647058,
        "width": "100%",
      },
      {
        "overflow": "hidden",
      },
      {
        "backgroundColor": "#DEF263",
      },
      {
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  testID="cryptoListObject"
>
  <Text
    style={
      [
        {
          "color": "#1414141F",
        },
        {
          "fontFamily": "Inter-Black",
        },
        {
          "fontSize": 150.4,
          "includeFontPadding": false,
          "letterSpacing": -2,
          "lineHeight": 150.4,
          "marginLeft": -4,
          "marginTop": -8,
        },
      ]
    }
  >
    BTC
  </Text>
  <View
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": "100%",
          "width": "100%",
        },
        {
          "justifyContent": "center",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 21.485714285714284,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
            {
              "marginLeft": 31.388235294117646,
            },
          ]
        }
      >
        test crypto
      </Text>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "justifyContent": "space-evenly",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "paddingRight": 31.388235294117646,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#141414",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        $1232134.33
      </Text>
      <Text
        style={
          [
            {
              "color": "#141414A3",
            },
            {
              "fontFamily": "Inter-SemiBold",
            },
            {
              "fontSize": 18.8,
              "includeFontPadding": false,
              "letterSpacing": 0,
            },
          ]
        }
      >
        12.123 BTC
      </Text>
    </View>
  </View>
</View>
`;
